digiroad2-api-oth/src/main/scala/fi/liikennevirasto/digiroad2/LaneApi.scala

```diff
@@ -0,0 +1,219 @@
+package fi.liikennevirasto.digiroad2
+
+import fi.liikennevirasto.digiroad2.Digiroad2Context.laneService.lanesWithConsistentRoadAddress
+import fi.liikennevirasto.digiroad2.Digiroad2Context.roadLinkService.{getRoadLinksAndChangesFromVVHWithPolygon, roadLinksWithConsistentAddress}
+import fi.liikennevirasto.digiroad2.Digiroad2Context.{laneService, roadAddressService, roadLinkService}
+import fi.liikennevirasto.digiroad2.client.VKMClient
+import fi.liikennevirasto.digiroad2.lane.LanePartitioner.{LaneWithContinuingLanes, getConnectedLanes, getStartingLanes}
+import fi.liikennevirasto.digiroad2.lane.{LanePartitioner, PieceWiseLane}
+import fi.liikennevirasto.digiroad2.linearasset.RoadLink
+import fi.liikennevirasto.digiroad2.util.{PolygonTools, RoadAddress, Track}
+import org.json4s.{DefaultFormats, Formats}
+import org.scalatra.{BadRequest, ScalatraServlet}
+import org.scalatra.json.JacksonJsonSupport
+import org.scalatra.swagger.{Swagger, SwaggerSupport}
+
+class LaneApi(val swagger: Swagger) extends ScalatraServlet with JacksonJsonSupport with SwaggerSupport {
+  lazy val vkmClient = new VKMClient
+  lazy val polygonTools = new PolygonTools
+
+  case class ApiRoad(roadNumber: Long, roadParts: Seq[ApiRoadPart])
+
+  case class ApiRoadPart(roadNumber: Option[Long], roadPartNumber: Long, apiLanes: Seq[ApiLane])
+
+  case class ApiLane(laneCode: Int, track: Int, startAddressM: Long, endAddressM: Long)
+
+  override protected def applicationDescription: String = "Lanes API"
+
+  override protected implicit def jsonFormats: Formats = DefaultFormats
+
+  //Description of Api entry point to get all lanes in specific road address range in road address format
+  val getLanesInRoadAddressRange =
+    (apiOperation[Long]("getLanesInRoadAddressRange")
+      .parameters(
+        queryParam[Int]("roadNumber").description("Road Number for the range"),
+        queryParam[Int]("track").description("Track code for search"),
+        queryParam[Int]("startRoadPartNumber").description("Starting road part number for search"),
+        queryParam[Int]("startAddrM").description("Starting distance on starting roadPart for search"),
+        queryParam[Int]("endRoadPartNumber").description("Ending road part number for search"),
+        queryParam[Int]("endAddrM").description("Ending distance on last road part for search"),
+        pathParam[String]("lanes_in_range").description("Get lanes in given range")
+      )
+      tags "LaneApi"
+      summary "Get lanes in given road address range in road address format"
+      authorizations "Contact your service provider for more information"
+      description "Example URL: /externalApi/lanes/lanes_in_range?road_number=9&track=1&start_part=208&start_addrm=8500&end_part=208&end_addrm=9000"
+      )
+
+  //Description of Api entry point to get all lanes in specific road address range
+  val getLanesInMunicipality =
+    (apiOperation[Long]("getLanesInMunicipality")
```

Comment:
näkyykö tämä ulospäin vai vain sisäisesti ? , dokumentaatio sanoisi että tää tulisi tohon summary osuuteen mutta alempana summary on erikseen määritetty